### ISSUE TITLE
emicorr memory and run time improvement

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8848]
_Issue [JP-3770](https://jira.stsci.edu/browse/JP-3770) was created on JIRA by [Maria Pena-Guerrero](https://jira.stsci.edu/secure/ViewProfile.jspa?name=pena):_

While working on optimizing Detector1 memory usage and runtime, I found that this small change made a big improvement for the performance of the emicorr step:

emicorr.py script, staring at line 441

 
```java
# Safety catch; anywhere the noise value is not finite, set it to zero 
dd_noise[~np.isfinite(dd_noise)] = 0.0

# Subtract EMI noise from the input data        
log.info('Subtracting EMI noise from data')

 # Interleave (straight copy) into 4 amps        
noise_x = np.arange(nx4) * 4        
for k in range(4):            
    output_model.data[..., noise_x + k] = output_model.data[..., noise_x + k] - dd_noise

```
then add the dd_noise array to the deletion statements

**Comment by stscijgbot-jp** [comment]
_Comment by [Maria Pena-Guerrero](https://jira.stsci.edu/secure/ViewProfile.jspa?name=pena) on [JIRA](https://jira.stsci.edu/browse/JP-3770?focusedCommentId=732349#comment-732349):_

Here are a couple of my test cases
|| || ||**Memory Used**|| || |**Step Time**| |
||File ||File size||Main||Branch|| ||Main||Branch||
|jw01386007001_04101_00006_mirimage_uncal.fits|306.5 MB|5 GB|4 GB| |88.6 sec|82.5 sec|
|jw01283001001_03101_00001_mirimage_uncal.fits|2.64 GB|48 GB|31 GB| |58.5 min|6.2 min|

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-3770?focusedCommentId=732534#comment-732534):_

Thanks [Maria Pena-Guerrero](https://jira.stsci.edu/secure/ViewProfile.jspa?name=pena) , that's useful to see your numbers, which are likely representative of most end users.  I think it also reinforces how tightly linked memory and runtime are; with 64 GB of installed RAM the runtime of even the jw01283001001_03101_00001_mirimage_uncal.fits example case is just 2 minutes (before and after), so we clearly take a large performance hit when swapping is required.

**Comment by stscijgbot-jp** [comment]
_Comment by [Tyler Pauly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tpauly) on [JIRA](https://jira.stsci.edu/browse/JP-3770?focusedCommentId=732585#comment-732585):_

Fixed by #8849
